src/misc.cpp

```diff
@@ -56,6 +58,41 @@ void InitializeCheats();
 void InitializeNPF();
 void InitializeOldNames();
 
+/**
+ * Generate an UUID.
+ *
+ * It isn't as much of an UUID as we would like, but our random generator can
+ * only produce 32bit random numbers.
+ * That is why we combine InteractiveRandom with the current (steady) clock.
+ * The first to add a bit of randomness, the second to ensure you can't get
```

Comment:
Technically the clock does not ensure it; if you achieve to start the same savegame within 1 millisecond...
In any case, the MD5 checksum afterwards undoes any ensuring you do before that.
Since both `InteractiveRandom` and `steady_clock` are essentially time based (`InteractiveRandom` gets seeded by time), it is plausible that two processes starting at the exact same time yield the same result. So I'd just phrase the comment slightly more cautiously. What about the following:
```
This doesn't create a UUID as per the specifications (RFC 4122). Not because it is complex,
but because it leaks identifiable information, i.e. the IEEE address / MAC address. This makes
the specific computer from which multiple reports were sent identifiable.
It does create a string of 32 hexadecimal digits that are based on the subject, current time,
and the `InteractiveRandom()`, which is a pseudo-random number generator seeded with the time
OpenTTD was started plus the number of times `InteractiveRandom()` was called. These inputs
make it highly unlikely that the same identifier is generated twice, while not containing any
identifiable information. 
Finally this data is passed through a MD5 hash to make it very difficult to get back the
original input adding another layer against potentially leaking identifiable information.
```

Reply:
I think you have been overthinking this, and I am not going to put that huge block of comments there :D

I changed UUID to "unique ID", and left most of the wording. The idea is just that this isn't as unique as we would like; ideally we switch to libuuid, but that is something for another PR :)

>  it is plausible that two processes starting at the exact same time yield the same result

That is very much not true. It is VERY unlikely this happens. Being able to load a savegame at the exact same millisecond (and now nanosecond) while you also started OpenTTD at the exact same time? Yeah, you are not going to make that in any realistic scenario :)
These kind of functions are not meant to avoid abuse (as you can just change the source code and make it return `42` at all times). They are merely meant that 2 users in different places in the world have a very very low probability to generate the same identifier :) I added that in the comment of the function now!

PS: the md5 is not meant to make this non-identifiable, as all inputs are already not identifiable. But it is meant to output a known-length string :) That is all :P